edgraph/server.go

```diff
@@ -698,10 +714,50 @@ func findVars(gmu *gql.Mutation) []string {
 	return varsList
 }
 
+// updateValMutations picks the val() from object and replaces it with its value
+// Assumption is that Subject should contain UID, whereas Object will VAL
+// If val(variable) exists in a query, but the values are not there for the variable,
+// it will delete the mutation itself, and stop it from execution
```

Comment:
instead of `stop it from execution`, you could write `ignore the mutation silently`